kong/cache_warmup.lua

```diff
@@ -128,4 +113,36 @@ function cache_warmup.execute(entities)
 end
 
 
+-- Warms DNS cache on select entities (currently only 'services')
+function cache_warmup.warm_services_dns()
+  if not kong.cache then
+    return true
+  end
+
+  local host_count = 0
+  local hosts_array = {}
+  local hosts_set = {}
+
+  for entity, err in kong.db.services:each(1000) do
```

Comment:
I think we’ve already paginated through all relevant services as part of the router rebuild, it would be less pressure on the database to have this function receive a set of hostnames, rather than generate it again without context.

Reply:
@p0pr0ck5 yes, there are 3 places:
1. router initial build (which happens with db on init, so cannot be used)
2. services warmup (the warmup in general is limited to worker=0, so could return the data, but makes interface a bit ugly).

Robert, if you have any time, I guess best way to bring this forward is to have some numbers. The numbers I have is that it makes any new request basically latency 0, also on router rebuilds. But does it really matter... perhaps not. Certainly may look better in some tests, but gains in a realworld are not that huge I guess, or are they?